src/EntityFramework6.Npgsql/NpgsqlMigrationSqlGenerator.cs

```diff
@@ -501,6 +499,21 @@ protected virtual void Convert(DropPrimaryKeyOperation dropPrimaryKeyOperation)
 
         #region Misc functions
 
+        /// <summary>
+        /// Quotes an identifier for Postgres SQL
+        /// </summary>
+        /// <param name="identifier"> The identifier to be quoted. </param>
+        /// <returns> The quoted identifier. </returns>
+        protected virtual string QuoteIdentifier(string identifier)
+        {
+            if (identifier == null)
+            {
+                throw new ArgumentNullException(nameof(identifier));
+            }
+
+            return '"' + identifier.Replace("\"", "") + '"';
```

Comment:
This creates unnecessary allocations. Instead of using ```Replace```, consider just doing a quick check if the first and last chars are the double quote and returning the original as-is.

Also, since the most common path is always a StringBuilder trying to append, pass the StringBuilder from the caller to this method and append to its buffer instead of using string concatenation.